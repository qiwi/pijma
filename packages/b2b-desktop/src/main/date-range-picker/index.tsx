import { Icon, Overlay, styled } from '@qiwi/pijma-core'
import { TextField } from '@qiwi/pijma-desktop'
import { format } from 'date-fns'
import React, { Component } from 'react'
import DayPicker from 'react-day-picker'
import { WithTranslation, withTranslation } from 'react-i18next'

import {
  getMonths,
  getWeekDaysLong,
  getWeekDaysShort,
} from '../date-picker/locale'
import Wrap from '../date-picker/wrap'
import { COLOR } from '../theme'
import DateRangePickerState from './DateRangePickerState'
import DateRangerPickerProps from './DateRangerPickerProps'

export const PickerDropdown = styled('div')`
  position: absolute;
  background: #fff;
  z-index: 10;
  width: 596px;
  border-radius: 8px;
  box-shadow: ${COLOR.SHADOW.Z3};

  .Selectable .DayPicker-Day {
    border-radius: 0 !important;
  }
  .Selectable .DayPicker-Day--start {
    border-top-left-radius: 50% !important;
    border-bottom-left-radius: 50% !important;
  }
  .Selectable .DayPicker-Day--end {
    border-top-right-radius: 50% !important;
    border-bottom-right-radius: 50% !important;
  }
  .Selectable
    .DayPicker-Day--selected:not(.DayPicker-Day--start):not(.DayPicker-Day--end):not(.DayPicker-Day--outside) {
    background-color: #ff8c0066 !important;
    color: #000000;
  }
  .DayPicker-NavButton {
    margin: 0 0;
    right: 1.5rem;
  }
  .DayPicker-NavButton--prev {
    margin-right: 2rem;
  }
`

export const PickerCellContainer = styled('div')`
  width: 30px;
  height: 30px;
  display: flex;
  align-items: center;
  justify-content: center;
`

const Container = styled('div')`
  position: relative;
  width: 100%;
`

const RedHelpText = styled('div')`
  color: ${COLOR.TEXT.Error};
`

const InputWrapper = styled('div')`
  input {
    color: transparent;
    text-shadow: 0 0 0 #000;
  }
`

class DisabledInputDateRangePicker extends Component<
  DateRangerPickerProps & WithTranslation,
  DateRangePickerState
> {
  static defaultProps = {
    numberOfMonths: 2,
  }

  myRef: any
  inputRef: any

  constructor(props: DateRangerPickerProps & WithTranslation) {
    super(props)
    this.state = {
      showPicker: false,
      errorDate: false,
      helpText: null,
    }
    this.myRef = React.createRef()
    this.inputRef = React.createRef()
    this._timeoutID = -1
  }

  _timeoutID: number

  formatPeriod = (dateFrom: Date, dateTo: Date) => {
    if (!dateFrom) {
      return ''
    }

    if (
      !dateTo ||
      format(dateFrom, 'dd.MM.yyyy') === format(dateTo, 'dd.MM.yyyy')
    ) {
      return format(dateFrom, 'dd.MM.yyyy')
    }

    return `${format(dateFrom, 'dd.MM.yyyy')} - ${format(dateTo, 'dd.MM.yyyy')}`
  }

  handleDayClick = (date: any, { disabled }: any) => {
    if (disabled) {
      return
    }

    const normalizeDate = {
      from: this.props.dateFrom,
      to: this.props.dateTo || this.props.dateFrom,
    }

    const range = DayPicker.addToRange(date, normalizeDate)
    const { from: dateFrom, to: dateTo } = range || normalizeDate

    if ((dateFrom || dateTo) && this.props.onChange) {
      this.props.onChange({ dateFrom, dateTo: dateTo || dateFrom })
    }
  }

  onFocus = () => {
    clearTimeout(this._timeoutID)
    this.setState({ showPicker: true })
  }

  onBlur = () => {
    this._timeoutID = window.setTimeout(() => {
      this.setState({ showPicker: false })
    })
  }

  onIconClick = () => {
    this.inputRef.current.focus()
  }

  render() {
    const { t } = this.props
    const MONTHS = getMonths(t)
    const WEEKDAYS_LONG = getWeekDaysLong(t)
    const WEEKDAYS_SHORT = getWeekDaysShort(t)
    const { errorDate, helpText, showPicker } = this.state
    const {
      minDate,
      maxDate,
      dayPickerProps,
      title,
      numberOfMonths,
      dateTo,
      dateFrom,
      inputValue,
    } = this.props
    const modifiers = { start: dateFrom, end: dateTo }
    const disabled: any = {}
    const selectedDays = [dateFrom, { from: dateFrom, to: dateTo }]
    const value = inputValue || this.formatPeriod(dateFrom, dateTo)

    if (minDate) {
      disabled.before = minDate
    }
    if (maxDate) {
      disabled.after = maxDate
    }

    return (
      <Wrap
        onFocus={this.onFocus}
        onBlur={this.onBlur}
        ref={this.inputRef}
        tabIndex={0}
      >
        <InputWrapper>
          <TextField
            type="text"
            title={title || t('datePicker:date')}
            hint={
              <div onClick={this.onIconClick}>
                <Icon name="calendar" />
              </div>
            }
            onChange={() => {
              /* noop */
            }}
            value={value}
            error={errorDate ? t('datePicker:wrongDate') : undefined}
            help={<RedHelpText>{helpText}</RedHelpText>}
          />
        </InputWrapper>
        <Container ref={this.myRef} />

        <Overlay
          show={showPicker}
          placement="bottom"
          container={this.myRef.current}
          target={this.myRef.current}
        >
          {() => (
            <PickerDropdown onFocus={this.onFocus}>
              {/* @ts-ignore */}
              <DayPicker
                className="Selectable"
                disabledDays={disabled}
                months={MONTHS}
                weekdaysLong={WEEKDAYS_LONG}
                weekdaysShort={WEEKDAYS_SHORT}
                // @ts-ignore
                renderDay={(day) => (
                  <PickerCellContainer>
                    <div>{day.getDate()}</div>
                  </PickerCellContainer>
                )}
                numberOfMonths={numberOfMonths}
                selectedDays={selectedDays}
                modifiers={modifiers}
                onDayClick={this.handleDayClick}
                firstDayOfWeek={1}
                {...dayPickerProps}
              />
            </PickerDropdown>
          )}
        </Overlay>
      </Wrap>
    )
  }
}

const DateRangePicker = withTranslation()(DisabledInputDateRangePicker)

export { DateRangePicker }
