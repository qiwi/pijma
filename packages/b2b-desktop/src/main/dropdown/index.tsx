import { Overlay, styled } from '@qiwi/pijma-core'
import React, { Component, Fragment } from 'react'

import { Container } from './Container'
import DropdownProps from './DropdownProps'

export const ContainerModal = styled('div')`
  position: relative;
  width: 100%;
`

class Dropdown extends Component<DropdownProps> {
  myRef: any

  constructor(props: DropdownProps) {
    super(props)
    this.myRef = React.createRef()
    this.onSelect = this.onSelect.bind(this)
  }

  onSelect(el: any) {
    if (this.props.onChange) {
      if (this.props.multiSelect && this.props.focus) {
        this.props.focus()
      }
      this.props.onChange(el)
    }
  }

  render() {
    const { value, items, show, multiSelect, renderItem } = this.props

    return (
      <Fragment>
        <ContainerModal ref={this.myRef} />
        <Overlay
          show={show}
          placement="bottom"
          container={this.myRef.current}
          target={this.myRef.current}
        >
          {() => (
            <Container
              multiSelect={multiSelect}
              items={items}
              value={value}
              onSelect={this.onSelect}
              renderItem={renderItem}
            />
          )}
        </Overlay>
      </Fragment>
    )
  }
}

export { Dropdown, Container }
