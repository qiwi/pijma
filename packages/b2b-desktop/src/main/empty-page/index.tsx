import { Card, Flex, FlexItem } from '@qiwi/pijma-core'
import { Heading, Paragraph } from '@qiwi/pijma-desktop'
import React, { Component } from 'react'

import { B2bIcon } from '../icon'
import { Offset } from '../offset'

class EmptyPage extends Component {
  render() {
    return (
      <div>
        <Offset mb={24} mt={40} ml={0} mr={0}>
          <Heading size="1">Ошибка 404</Heading>
        </Offset>

        <Card
          r={10}
          width="1180px"
          height="500px"
          bg="#fff"
          p="44px 44px 48px 44px"
        >
          <Flex>
            <FlexItem mr={3}>
              <B2bIcon icon="attention" h={24} w={24} iconColor="#d0021b" />
            </FlexItem>
            <FlexItem>
              <Offset mb={8} ml={0} mr={0}>
                <Heading size="4">Страница не найдена</Heading>
              </Offset>
              <Paragraph bold={false} size="m" color="support">
                Возможно, в вашей ссылке опечатка.
              </Paragraph>
              <Paragraph bold={false} size="m" color="support">
                Или когда-то такая страница на сайте была, но теперь её нет.
              </Paragraph>
            </FlexItem>
          </Flex>
        </Card>
      </div>
    )
  }
}

export default EmptyPage
