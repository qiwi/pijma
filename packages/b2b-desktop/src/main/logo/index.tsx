import { Box } from '@qiwi/pijma-core'
import React from 'react'

export const Logo = () => (
  <Box width={'266px'} height={'64px'}>
    <svg
      width="266"
      height="64"
      viewBox="0 0 266 64"
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g clipPath="url(#clip0_447_5)">
        <path d="M265.142 0H0V64H265.142V0Z" fill="white" />
        <path
          d="M78.5211 41.1624V22.7337H74.0112V41.1624H78.5211ZM56.7462 41.1624L59.0651 31.8579C59.2358 31.1454 59.3567 30.4219 59.4267 29.6926H59.4779C59.5472 30.4219 59.6677 31.1455 59.8384 31.8579L62.1542 41.1624H67.1803L71.8459 22.7337H67.1803L64.6544 34.2536C64.5197 34.89 64.4433 35.5373 64.4261 36.1875H64.3739C64.3332 35.5348 64.2296 34.8875 64.0645 34.2547L61.3072 22.7347H57.6688L54.8336 34.2547C54.6692 34.8877 54.5656 35.5349 54.5243 36.1875H54.4731V36.148C54.4507 35.5114 54.3732 34.878 54.2416 34.2547L51.9216 22.7347H47.1792L51.6592 41.1635L56.7462 41.1624ZM44.9061 41.1624V22.7337H40.4005V41.1624H44.9061ZM27.9248 37.348C25.0128 37.348 22.9509 34.9768 22.9509 31.8323C22.9509 28.8158 25.0128 26.548 27.9248 26.548C30.76 26.548 32.8997 28.8158 32.8997 31.8323C32.8993 32.6478 32.7327 33.4546 32.4101 34.2035L30.5797 32.4264L28.0528 35.2862L29.9344 36.9352C29.368 37.219 28.5947 37.348 27.9248 37.348ZM35.1163 41.7299L37.6677 38.868L35.7328 37.2958C36.8155 35.7502 37.5632 33.9454 37.5632 31.8323C37.5632 26.548 33.6165 22.4243 27.9227 22.4243C22.2011 22.4243 18.2832 26.548 18.2832 31.8323C18.2832 37.2446 22.2011 41.4718 27.9227 41.4718C29.9333 41.4718 31.7371 40.931 33.2325 39.9507L35.1163 41.7299Z"
          fill="#FF6F00"
        />
        <path
          d="M110.516 22.1376L100.658 31.9956L110.516 41.8536L120.374 31.9956L110.516 22.1376Z"
          fill="#FF6F00"
        />
        <path
          d="M110.517 41.8537L106.658 45.714H86.9492L100.659 31.9956L110.517 41.8537Z"
          fill="#FFB374"
        />
        <path
          d="M110.517 22.1374L100.659 31.9955L86.9492 18.2856H106.658L110.517 22.1374Z"
          fill="#FFB374"
        />
        <path
          d="M129.686 22.7412H136.279C139.641 22.7412 142.012 24.4617 142.012 27.5113C142.012 29.2308 141.179 30.7167 139.772 31.4729V31.5252C141.804 32.1247 142.769 34.0009 142.769 35.8783C142.769 39.5785 139.798 41.2468 136.383 41.2468H129.686V22.7412ZM136.305 30.3007C137.764 30.3007 138.599 29.2575 138.599 27.9284C138.599 26.5993 137.79 25.6351 136.252 25.6351H133.046V30.3007H136.305ZM136.643 38.3529C138.364 38.3529 139.328 37.2329 139.328 35.6692C139.328 34.1311 138.337 33.0111 136.643 33.0111H133.047V38.3529H136.643Z"
          fill="#272A35"
        />
        <path
          d="M145.828 22.7412H149.19V34.5748C149.19 37.0762 150.806 38.4842 153.282 38.4842C155.758 38.4842 157.374 37.0762 157.374 34.5748V22.7412H160.736V34.6015C160.736 38.7444 157.765 41.5594 153.309 41.5594C148.829 41.5594 145.828 38.7444 145.828 34.6015V22.7412Z"
          fill="#272A35"
        />
        <path
          d="M165.326 36.5034C166.42 37.4933 168.192 38.4575 169.913 38.4575C171.581 38.4575 172.623 37.5978 172.623 36.3466C172.623 32.8266 163.997 33.5583 163.997 27.7983C163.997 24.5396 166.707 22.4287 170.304 22.4287C172.233 22.4287 174.266 23.0804 175.725 24.4095L174.239 27.1466C173.223 26.2346 171.633 25.5306 170.174 25.5306C168.479 25.5306 167.359 26.3647 167.359 27.6415C167.359 31.0815 175.986 30.1962 175.986 36.0863C175.986 39.0837 173.796 41.5594 169.835 41.5594C167.54 41.5594 165.169 40.6997 163.502 39.0314L165.326 36.5034Z"
          fill="#272A35"
        />
        <path
          d="M179.396 22.7412H182.758V41.2468H179.396V22.7412Z"
          fill="#272A35"
        />
        <path
          d="M187.076 22.7412H190.437L197.448 33.4281C198.126 34.5225 198.987 36.2943 198.987 36.2943H199.038C199.038 36.2943 198.855 34.5225 198.855 33.4281V22.7412H202.192V41.2468H198.855L191.844 30.5865C191.141 29.4921 190.307 27.7204 190.307 27.7204H190.255C190.255 27.7204 190.437 29.4921 190.437 30.5865V41.2468H187.076V22.7412Z"
          fill="#272A35"
        />
        <path
          d="M206.561 22.7412H217.637V25.6351H209.919V30.4831H216.149V33.3759H209.919V38.3529H218.051V41.2468H206.561V22.7412Z"
          fill="#272A35"
        />
        <path
          d="M221.939 36.5034C223.034 37.4933 224.805 38.4575 226.526 38.4575C228.194 38.4575 229.236 37.5978 229.236 36.3466C229.236 32.8266 220.61 33.5583 220.61 27.7983C220.61 24.5396 223.321 22.4287 226.917 22.4287C228.846 22.4287 230.879 23.0804 232.338 24.4095L230.852 27.1466C229.836 26.2346 228.246 25.5306 226.787 25.5306C225.092 25.5306 223.972 26.3647 223.972 27.6415C223.972 31.0815 232.6 30.1962 232.6 36.0863C232.6 39.0837 230.41 41.5594 226.448 41.5594C224.154 41.5594 221.782 40.6997 220.115 39.0314L221.939 36.5034Z"
          fill="#272A35"
        />
        <path
          d="M236.197 36.5034C237.291 37.4933 239.063 38.4575 240.784 38.4575C242.452 38.4575 243.494 37.5978 243.494 36.3466C243.494 32.8266 234.868 33.5583 234.868 27.7983C234.868 24.5396 237.578 22.4287 241.175 22.4287C243.104 22.4287 245.137 23.0804 246.596 24.4095L245.11 27.1466C244.094 26.2346 242.504 25.5306 241.045 25.5306C239.35 25.5306 238.23 26.3647 238.23 27.6415C238.23 31.0815 246.857 30.1962 246.857 36.0863C246.857 39.0837 244.667 41.5594 240.706 41.5594C238.411 41.5594 236.04 40.6997 234.373 39.0314L236.197 36.5034Z"
          fill="#272A35"
        />
      </g>
      <defs>
        <clipPath id="clip0_447_5">
          <rect width="265.142" height="64" fill="white" />
        </clipPath>
      </defs>
    </svg>
  </Box>
)
