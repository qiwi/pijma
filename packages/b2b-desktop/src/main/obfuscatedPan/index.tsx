import React from 'react'

import ObfuscatedPanProps from './ObfuscatedPanProps'

export const ObfuscatedPan = ({ pan, panPrefix = '*' }: ObfuscatedPanProps) => {
  return (
    <span>
      {panPrefix}
      {pan ? pan.slice(-4) : null}
    </span>
  )
}
