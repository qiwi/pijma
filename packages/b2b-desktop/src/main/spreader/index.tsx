import { styled } from '@qiwi/pijma-core'

export default styled('div')`
  display: flex;
  flex-direction: column;
  margin: 0 auto;
  max-width: 1280px;
  min-width: 400px;
  > * {
    padding: 0 50px !important;
  }
`
