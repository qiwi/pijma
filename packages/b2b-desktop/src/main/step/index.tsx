import { ButtonControl, Typo } from '@qiwi/pijma-core'
import { Text } from '@qiwi/pijma-desktop'
import React, { FC } from 'react'

import StepProps from './StepProps'
import StepButton from './StepStyle'

const Step: FC<StepProps> = (props) => {
  const { icon, text, variant = 'default', ...rest } = props
  return (
    <ButtonControl {...rest}>
      {({ active, focus, hover, ...events }) => (
        <StepButton variant={variant} {...events}>
          <Text>
            <Typo css={{ overflowWrap: 'normal' }}>{icon || text}</Typo>
          </Text>
        </StepButton>
      )}
    </ButtonControl>
  )
}

export { Step }
