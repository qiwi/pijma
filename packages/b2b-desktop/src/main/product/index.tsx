import { Flex, Icon, styled } from '@qiwi/pijma-core'
import React, { Component } from 'react'

import { Dropdown } from '../dropdown'
import { COLOR } from '../theme'
import ProductProps from './ProductProps'
import ProductState from './ProductState'

const ProductType = styled('div')`
  font-weight: 500;
  font-size: 14px;
  color: ${COLOR.TEXT.Primary};
`

const ProductMerchant = styled('div')`
  font-size: 12px;
  font-weight: 300;
  color: ${COLOR.TEXT.Secondary2};
`

const Container = styled('div')`
  color: black;
  position: relative;
  width: 100%;
  top: -10px;
  left: -1px;
`

const IconWrapper = styled('div')`
  width: 24px;
  height: 24px;
  display: inline-block;
`

const ProductWrapper = styled('div')``

export class Product extends Component<ProductProps, ProductState> {
  _timeoutID: any
  constructor(props: ProductProps) {
    super(props)
    this.state = {
      isOpen: false,
    }
    this.onBlur = this.onBlur.bind(this)
    this.onFocus = this.onFocus.bind(this)
    this._timeoutID = -1
  }

  onFocus() {
    clearTimeout(this._timeoutID)
    this.setState({ isOpen: !this.state.isOpen })
  }

  onBlur() {
    this._timeoutID = setTimeout(() => {
      this.setState({ isOpen: false })
    }, 10)
  }

  render() {
    const { selectorData, title, onChange } = this.props
    const isShownDropdown = selectorData.items
      ? selectorData.items.length > 1
      : null

    return (
      <ProductWrapper>
        <ProductType>{title}</ProductType>
        <ProductMerchant
          tabIndex={0}
          onClick={this.onFocus}
          onBlur={this.onBlur}
        >
          <Flex align="center">
            <span>{selectorData.currentItem}</span>
            {isShownDropdown && (
              <IconWrapper>
                <Icon name="angle-small-down" />
              </IconWrapper>
            )}
          </Flex>
        </ProductMerchant>
        {isShownDropdown && (
          <Container>
            <Dropdown
              show={this.state.isOpen}
              items={selectorData.items || []}
              value={selectorData.currentItem}
              onChange={onChange}
            />
          </Container>
        )}
      </ProductWrapper>
    )
  }
}
