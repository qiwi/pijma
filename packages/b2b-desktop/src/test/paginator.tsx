import React from 'react'
import renderer from 'react-test-renderer'

import { Paginator } from '../main'

describe('Paginator', () => {
  it('is properly exported from lib index', () => {
    expect(Paginator).not.toBeUndefined()
  })

  it('renders correctly', () => {
    const paginator = renderer
      .create(
        <Paginator
          activePage={2}
          totalItemsCount={40}
          pageRangeDisplayed={5}
          itemsCountPerPage={10}
        />,
      )
      .toJSON()
    expect(paginator).toMatchSnapshot()
  })
})
