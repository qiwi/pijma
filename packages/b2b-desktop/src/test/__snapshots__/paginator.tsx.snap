// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paginator renders correctly 1`] = `
<div
  className="css-1wm1qhp"
>
  <ul
    className="pagination"
  >
    <li
      className=""
      onClick={[Function]}
    >
      <a
        aria-label="Go to page number 1"
        className=""
        href="#"
      >
        1
      </a>
    </li>
    <li
      className="active"
      onClick={[Function]}
    >
      <a
        aria-label="Go to page number 2"
        className="undefined"
        href="#"
      >
        2
      </a>
    </li>
    <li
      className=""
      onClick={[Function]}
    >
      <a
        aria-label="Go to page number 3"
        className=""
        href="#"
      >
        3
      </a>
    </li>
    <li
      className=""
      onClick={[Function]}
    >
      <a
        aria-label="Go to page number 4"
        className=""
        href="#"
      >
        4
      </a>
    </li>
  </ul>
</div>
`;
